extensions/blocks/tiled-gallery/tiled-gallery.php

```diff
@@ -26,14 +28,85 @@ class_exists( 'Jetpack_Photon' ) && Jetpack::is_module_active( 'photon' )
 	 *
 	 * @return string
 	 */
-	function jetpack_tiled_gallery_load_block_assets( $attr, $content ) {
+	public static function render( $attr, $content ) {
 		$dependencies = array(
 			'lodash',
 			'wp-i18n',
 			'wp-token-list',
 		);
 		Jetpack_Gutenberg::load_assets_as_required( 'tiled-gallery', $dependencies );
 
+		$is_squareish_layout = self::is_squareish_layout( $attr );
+		$srcset_step         = 300;
+
+		if ( function_exists( 'jetpack_photon_url' )
+			&& preg_match_all( '/<img [^>]+>/', $content, $images )
+		) {
+			/**
+			 * This block processes all of the images that are found and builds $find and $replace.
+			 *
+			 * The original img is added to the $find array and the replacement is made and added
+			 * to the $replace array. This is so that the same find and replace operations can be
+			 * made on the entire $content.
+			 */
+			$find    = array();
+			$replace = array();
+
+			foreach ( $images[0] as $image_html ) {
+				if (
+					preg_match( '/data-width="([0-9]+)"/', $image_html, $img_height )
+					&& preg_match( '/data-height="([0-9]+)"/', $image_html, $img_width )
+					&& preg_match( '/src="([^"]+)"/', $image_html, $img_src )
+				) {
+					$orig_width   = $img_width[1];
+					$orig_height  = $img_height[1];
+					$orig_src     = $img_src[1];
+					$srcset_parts = array();
+
+					if ( $is_squareish_layout ) {
+						$min_width = min( 600, $orig_width, $orig_height );
+						$max_width = min( 2000, $orig_width, $orig_height );
```

Comment:
I don't feel strongly about this but we _could_ wrap this with core filter to let sites limit srcset max size, see:

https://github.com/WordPress/WordPress/blob/adc7f9fe544b2c1c9ebb71d9fc0206fe39d6480c/wp-includes/media.php#L1115-L1123

Reply:
Initially, that seems reasonable. However, upon looking at the filter a bit closer i'm not clear it's a good fit here:

```php
	/**
	 * Filters the maximum image width to be included in a 'srcset' attribute.
	 *
	 * @since 4.4.0
	 *
	 * @param int   $max_width  The maximum image width to be included in the 'srcset'. Default '1600'.
	 * @param array $size_array Array of width and height values in pixels (in that order).
	 */
	$max_srcset_image_width = apply_filters( 'max_srcset_image_width', 1600, $size_array );
```

We need to pass an array of image sizes? Why?

Mostly given the uncertainty about this and the fact that code freeze is close, I'd like to hold off on this for now.